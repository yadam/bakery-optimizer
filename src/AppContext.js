import PropTypes from 'prop-types';
import React, { useState } from 'react';

export const AppContext = React.createContext();

export const AppContextProvider = ({ children }) => {
  const [breads, setBreads] = useState([]);
  const [customers, setCustomers] = useState([]);
  return (
    <AppContext.Provider value={{ breads, customers, setBreads, setCustomers }}>
      {children}
    </AppContext.Provider>
  );
};

AppContextProvider.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.arrayOf(PropTypes.node),
  ]),
};

AppContextProvider.defaultProps = {
  children: undefined,
};

export default { AppContext, AppContextProvider };
