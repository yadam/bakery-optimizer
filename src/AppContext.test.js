import React from 'react';
import renderer from 'react-test-renderer';
import { AppContextProvider } from './AppContext';

describe('AppContextProvider', () => {
  it('renders with children', () => {
    const tree = renderer
      .create(
        <AppContextProvider>
          <div>
            <span>Hello World!</span>
          </div>
        </AppContextProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders without children', () => {
    const tree = renderer.create(<AppContextProvider />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
