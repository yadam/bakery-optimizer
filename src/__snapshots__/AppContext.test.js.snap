// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppContextProvider renders with children 1`] = `
<div>
  <span>
    Hello World!
  </span>
</div>
`;

exports[`AppContextProvider renders without children 1`] = `null`;
